### DISCUSSION TITLE
Do linked sets have to be distinct?

### DISCUSSION THREAD

**Discussion by amaster507** [geldata/gel#3425] [Q&A] [answered]
Going through the Tutorial, and I come to:

> However, an important thing to remember is that following links always results in a set of objects (without duplicates), unlike properties.

So this means that if a Car is linked to the same Tire 4 times, when you query for `SELECT Car.tires { partNumbe }` you will only get the part number one time?

How can you show that the car needs 4 tires? Newb working through tutorial and only this far in so far, maybe this is answered later?

**Comment by elprans** [comment, marked as answer]
A link represents a semantic relationship between two entities.  It says "a Car has Tire" as it's part, and having four identical links between the same two objects would be like repeating "a Car has Tire" four times, i.e it's the same meaning.  What you are looking for is to specify some _details_ on how a Car and a specific Tire part are related.  This is where you can use link properties:

```
type Car {
  property model -> str;
  multi link parts -> Part {
    property count -> int64;
  }
}

abstract type Part { 
  property number -> str;
}

type Tire extending Part;
```

and then

```
insert Car {
  model := 'cybertruck', 
  parts := (
    select Tire {
      @count := 4  # <- this is where you specify the tire count
    } 
    filter .number = 'supertire no. 1'
  )
}
```

What this model gives you is the ability to quickly tell how many distinct kinds of parts a car has:

```
with car := (select Car filter .model = 'cybertruck')
select count(car.parts)
```

and if you want a table of parts with counts you can do:

```
with car := (select Car filter .model = 'cybertruck')
select (car.parts.number, car.parts@count)
```

**Reply by elprans** [reply]
Oh, and also if you want to get a set of all makes using a part you can do

```
with part := (select Part filter .number = 'supertire no. 1')
select part.<parts[is Car]
```

If links weren't unique, then you'd need to remember to place `distinct` in front of every such traversal, which would generally be error-prone and slow.

**Reply by amaster507** [reply]
thank you, this makes sense and I am use to only being able to have sets everywhere in the database with Dgraph. All lists are sets in Dgraph so you can never have a list of duplicates ever. I was just trying to understand limitations early on my journey.

I was thinking originally I could use the link property to define the location of each of the four parts, but that does not seem possible unless there was a linking type in the middle that negated the need for uniqueness but add an additional depth to the model. Car.partLocation.part where each PartLocation is unique but the Parts linked to each location would not have to be unique. This is also what I am use to doing already anyways.

Thanks again! Looking forward to the launch tomorrow!
